lib/ansible/module_utils/common/dict_transformations.py

```diff
@@ -123,6 +125,8 @@ def dict_merge(a, b):
 
 
 def recursive_diff(dict1, dict2):
```

Comment:
Maybe add a doc string as well.
```suggestion
def recursive_diff(dict1, dict2):
    """Recursively diff two dictionaries

    Raises ``TypeError`` for incorrect argument type.

    :arg dict1: Dictionary to compare against.
    :arg dict2: Dictionary to compare with ``dict1``.
    :return: Tuple of dictionaries of differences or ``None`` if there are no differences.
    """
```